File: python/fundamentals/day-03/MiniChallenge.py
import math

# 1. Ask for user input
num_input = int(input("Enter a positive integer: "))
num = abs(num_input)  # Work with positive value

print("\n--- NUMBER ANALYSIS SYSTEM ---")

# 2. Number of Digits
num_digits = len(str(num))
print(f"Number of digits: {num_digits}")

# 3. Sum of Digits
digit_sum = sum(int(digit) for digit in str(num))
print(f"Sum of digits   : {digit_sum}")

# 4. Reverse
reversed_str = str(num)[::-1]
reversed_num = int(reversed_str)
print(f"Reverse         : {reversed_num}")

# 5. Palindrome Check
is_palindrome = "Yes" if str(num) == reversed_str else "No"
print(f"Palindrome      : {is_palindrome}")

# 6. Factorial (Practical limit: <= 20 to keep console clean, though Python supports large numbers)
if num <= 20:
    print(f"Factorial       : {math.factorial(num)}")
else:
    print(f"Factorial       : Skipping (Number too large, max recommended is 20)")

# 7. Multiplication Table (1–10)
print("\nMultiplication Table (1–10):")
print("-" * 25)
for i in range(1, 11):
    print(f"{num} × {i:2d} = {num * i}")

File: java/fundamentals/day03/Minichallenge.java
package fundamentals.day03;
import java.util.Scanner;
import java.math.BigInteger;
public class Minichallenge {
	    public static void main(String[] args) {
	        Scanner scanner = new Scanner(System.in);

	        // 1. Ask for user input
	        System.out.print("Enter a positive integer: ");
	        long num = Math.abs(scanner.nextLong());

	        System.out.println("\n--- NUMBER ANALYSIS SYSTEM ---");

	        // 2. Number of Digits
	        String numStr = Long.toString(num);
	        int numDigits = numStr.length();
	        System.out.println("Number of digits: " + numDigits);

	        // 3. Sum of Digits
	        long temp = num;
	        long digitSum = 0;
	        while (temp > 0) {
	            digitSum += temp % 10;
	            temp /= 10;
	        }
	        System.out.println("Sum of digits   : " + digitSum);

	        // 4. Reverse
	        String reversedStr = new StringBuilder(numStr).reverse().toString();
	        long reversedNum = Long.parseLong(reversedStr);
	        System.out.println("Reverse         : " + reversedNum);

	        // 5. Palindrome Check
	        String isPalindrome = numStr.equals(reversedStr) ? "Yes" : "No";
	        System.out.println("Palindrome      : " + isPalindrome);

	        // 6. Factorial (Practical limit: <= 20 for standard 'long' storage)
	        if (num <= 20) {
	            long factorial = 1;
	            for (int i = 1; i <= num; i++) {
	                factorial *= i;
	            }
	            System.out.println("Factorial       : " + factorial);
	        } else {
	            System.out.println("Factorial       : Skipping (Number too large, max limit is 20 for 'long')");
	        }

	        // 7. Multiplication Table (1–10)
	        System.out.println("\nMultiplication Table (1–10):");
	        System.out.println("-------------------------");
	        for (int i = 1; i <= 10; i++) {
	            System.out.printf("%d × %2d = %d%n", num, i, num * i);
	        }

	        scanner.close();
	    }
	}

